src/opnsense/mvc/app/views/OPNsense/Core/firmware.volt

```diff
@@ -334,6 +334,15 @@
                         $('#updatestatus').html("{{ lang._('Audit done.') }}");
                     } else if ($.upgrade_action == 'action') {
                         $('#updatestatus').html("{{ lang._('Action done.') }}");
+                        $.ajax({
+                            type: 'GET',
+                            url: '/ui/core/firmware',
+                            dataType:'html',
+                            contentType: 'text/html',
+                            complete: function(data,status) {
+                                $('#mainmenu').html($('#mainmenu', data.responseText));
+                            }
+                        });
                     } else {
                         $('#updatestatus').html("{{ lang._('Upgrade done.') }}");
```

Comment:
Idea is good, but we need to do it for this line as well or better yet the whole upper else to catch all future cases. Some updates will be spurious, but better than not doing it.

Reply:
It seems I misread in my first response, but replacing the menu like this might have side affects on already registered events. 

Preventing users from going to going to non existing location isn't too hard to handle by requesting the current items from the api and removing the ones that are gone. Ensuring the menu is ajax enabled is a different story when doing it properly (its now a hybrid structure).